components/Skeleton/Skeleton.jsx

```diff
@@ -1,85 +1,45 @@
 import PropTypes from 'prop-types';
-// eslint-disable-next-line import/no-cycle
-import SkeletonTypes from './SkeletonTypes';
-import { spacing, baseFontSize as defaultBaseFontSize } from '../shared/theme';
-import {
-  SkeletonRect,
-  SkeletonCircle,
-  SkeletonText,
-  SkeletonButton,
-  SkeletonTag,
-} from './sub-components';
+import * as Styles from './Skeleton.styles';
 
-const Skeleton = props => {
-  const { type, size, theme, height, width } = props;
+const Skeleton = ({ type, ...props }) => {
+  switch (type) {
+    case 'circle':
+      return <Styles.Circle {...props} />;
 
-  const rectProps = {
-    height,
-    width,
-  };
-  const circleProps = {
-    height,
-    width,
-  };
+    case 'text':
+      return <Styles.Text {...props} />;
 
-  const textProps = {
-    height,
-    width,
-  };
+    case 'button':
+      return <Styles.Button {...props} />;
 
-  const buttonProps = {
-    size,
-    theme,
-    width,
-  };
+    case 'tag':
+      return <Styles.Tag {...props} />;
 
-  const tagProps = {
-    size,
-    theme,
-    width,
-  };
-
-  return (
-    <>
-      {type === 'rect' && <SkeletonRect {...rectProps} />}
-      {type === 'circle' && <SkeletonCircle {...circleProps} />}
-      {type === 'text' && <SkeletonText {...textProps} />}
-      {type === 'button' && <SkeletonButton {...buttonProps} />}
-      {type === 'tag' && <SkeletonTag {...tagProps} />}
-    </>
-  );
+    default:
+      return <Styles.Rect {...props} />;
+  }
 };
 
 Skeleton.defaultProps = {
   type: 'rect',
+  as: 'div',
   width: '100%',
   height: '14px',
   size: 'medium',
-  theme: {
-    baseFontSize: defaultBaseFontSize,
-    spacing,
-  },
 };
 
 Skeleton.propTypes = {
-  /** Sets the component behavior */
   type: PropTypes.oneOf(['rect', 'circle', 'text', 'button', 'tag']),
+  as: PropTypes.oneOf(['div', 'span']),
   width: PropTypes.string,
-  /** Will affect only types that not uses the prop `size` */
   height: PropTypes.string,
-  /** Will affect only `Tag` and `Button` types.  */
   size: PropTypes.oneOf(['xsmall', 'small', 'medium', 'large']),
-  /** Used only for themification.  */
-  theme: PropTypes.shape({
-    spacing: PropTypes.object,
-    baseFontSize: PropTypes.number,
-  }),
```

Comment:
Why shouldn't we be able to send a custom theme to the component?

Reply:
It's not exactly that we shouldn't. Let me try to explain my reasoning for removing this property.

We currently use the theme only to derive other values, as shown in these examples:

Example 1:

```ts
  height: ${({ size }) => {
    const sizes = {
      small: baseFontSize * 0.875,
      medium: baseFontSize * 1.125,
      large: baseFontSize * 1.25,
    };

    return `${(sizes[size] || sizes.medium) + spacing.xxsmall * 2}px`;
  }};
```

Example 2:

```ts
 height: ${({ size }) => {
    const sizes = {
      xsmall: spacing.large,
      small: spacing.xlarge,
      medium: spacing.xxlarge,
      large: spacing.xxxlarge,
    };

    return `${sizes[size] || sizes.medium}px`;
  }};
```

In this situation, I believe that using the theme as a way to "extend" this component may not be the best approach. It could be challenging for users to effectively update these values using the theme. 

If you think it's worth keeping this prop, I'm open to updating the pull request accordingly.

Reply:
This issue involving the best use of theme as the default value for styling is already on our "radar". However, we didn't have a moment to think about effective solutions for this case. Regarding the last point mentioned by you, I believe that you removed the thema prop due to the overwriting of the default value of the props passed to the component by importing the thema and direct use in the component's styling, which consequently ends up overwriting the value of thema passed by the user and, with that, making its use unfeasible. However, I believe it would be interesting to keep the thema props and just adjust their use in order to avoid the previously mentioned problem, what do you think about that?

Reply:
I think it's understandable